ui/vty-lib/src/Ariadne/UI/Vty/Widget/Wallet.hs

```diff
@@ -311,11 +318,13 @@ handleWalletWidgetEvent = \case
           walletNewAccountResultL .= NewAccountResultNone
         walletInfoL .= Just newInfo
 
-        walletNameL .= fromMaybe "" uwiLabel
         walletIdL .= uwiId
-        walletAccountsL .= map
-          (\(idx, UiAccountInfo{..}) -> WalletAccount idx (fromMaybe "" uaciLabel) uaciBalance False)
-          (zip [0..] uwiAccounts)
+        whenJust uwiLabel $ updateEditable walletNameL walletNameEditL id
+        let convert =
+                map (\(idx, UiAccountInfo{..}) ->
+                  WalletAccount idx (fromMaybe "" uaciLabel) uaciBalance False)
+              . zip [0..]
+        updateEditable walletAccountsL walletAccountsEditL convert uwiAccounts
```

Comment:
I'd rather store `[WalletAccount]` in `walletAccountsL` and keep `updateEditable` free of conversion logic.